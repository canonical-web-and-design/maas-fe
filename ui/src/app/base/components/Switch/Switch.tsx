import classNames from "classnames";
import PropTypes from "prop-types";
import React from "react";

type Props = {
  className?: string;
  // TODO: Investigate why this won't work with React.HTMLProps<HTMLInputElement>.
} & React.PropsWithoutRef<JSX.IntrinsicElements["input"]>;

const Switch = ({ className, ...inputProps }: Props): JSX.Element => {
  return (
    <span>
      <input
        className={classNames("p-switch", className)}
        type="checkbox"
        {...inputProps}
      />
      <div className="p-switch__slider"></div>
    </span>
  );
};

Switch.propTypes = {
  /**
   * Optional classes applied to the parent element.
   */
  className: PropTypes.string,
};

export default Switch;
