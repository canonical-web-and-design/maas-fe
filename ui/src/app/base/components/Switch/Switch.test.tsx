import { shallow } from "enzyme";
import React from "react";

import Switch from "./Switch";

describe("Switch", () => {
  it("renders", () => {
    const wrapper = shallow(<Switch />);

    expect(wrapper).toMatchSnapshot();
  });
});
