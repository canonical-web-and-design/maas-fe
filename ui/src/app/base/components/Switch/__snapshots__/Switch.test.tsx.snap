// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch renders 1`] = `
<span>
  <input
    className="p-switch"
    type="checkbox"
  />
  <div
    className="p-switch__slider"
  />
</span>
`;
